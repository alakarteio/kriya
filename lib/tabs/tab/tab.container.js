'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.mapDispatchToProps = exports.mapStateToProps = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _reactRedux = require('react-redux');

var _redux = require('../redux');

var _tab = require('./tab');

var _tab2 = _interopRequireDefault(_tab);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var mapStateToProps = exports.mapStateToProps = function mapStateToProps(state, _ref) {
  var code = _ref.code;

  return _extends({
    activated: (0, _redux.isActivated)(state)(code)
  }, (0, _redux.getTab)(state)(code));
};

var mapDispatchToProps = exports.mapDispatchToProps = function mapDispatchToProps(dispatch, _ref2) {
  var code = _ref2.code;

  return {
    onClick: function onClick() {
      return dispatch({ type: 'TAB_CLICKED', payload: code });
    }
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(_tab2.default);