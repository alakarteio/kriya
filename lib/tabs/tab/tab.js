'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _recompose = require('recompose');

var _reactInk = require('react-ink');

var _reactInk2 = _interopRequireDefault(_reactInk);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _tabStyles = require('../../../src/tabs/tab/tab.styles.scss');

var _tabStyles2 = _interopRequireDefault(_tabStyles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Tab = function Tab(_ref) {
  var style = _ref.style,
      className = _ref.className,
      label = _ref.label,
      activated = _ref.activated,
      onClick = _ref.onClick;

  var classes = (0, _classnames3.default)(_tabStyles2.default.tab, className, _defineProperty({}, _tabStyles2.default.activated, activated));

  return _react2.default.createElement(
    'div',
    { style: style, className: classes, onClick: onClick },
    _react2.default.createElement(_reactInk2.default, { style: { zIndex: 2 } }),
    label,
    activated && _react2.default.createElement('div', { className: _tabStyles2.default.border })
  );
};

Tab.propTypes = {
  style: _propTypes2.default.object,
  className: _propTypes2.default.string,
  label: _propTypes2.default.string,
  activated: _propTypes2.default.bool,
  onClick: _propTypes2.default.func.isRequired
};

Tab.defaultProps = {
  style: {},
  className: '',
  label: '',
  activated: false
};

exports.default = (0, _recompose.onlyUpdateForPropTypes)(Tab);