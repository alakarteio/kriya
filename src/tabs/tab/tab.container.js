import { connect } from 'react-redux'
import { isActivated, getTab } from '../redux'
import Component from './tab'

export const mapStateToProps = (state, { code }) => {
  return {
    activated: isActivated(state)(code),
    ...getTab(state)(code),
  }
}

export const mapDispatchToProps = (dispatch, { code }) => {
  return {
    onClick: () => dispatch({ type: 'TAB_CLICKED', payload: code }),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Component)
