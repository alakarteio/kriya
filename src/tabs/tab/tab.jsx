import React from 'react'
import PropTypes from 'prop-types'
import { onlyUpdateForPropTypes } from 'recompose'
import Ink from 'react-ink'
import classnames from 'classnames'
import styles from '../../../src/tabs/tab/tab.styles.scss'

const Tab = ({ style, className, label, activated, onClick }) => {
  const classes = classnames(
    styles.tab,
    className,
    {
      [styles.activated]: activated,
    },
  )

  return (
    <div style={style} className={classes} onClick={onClick}>
      <Ink style={{ zIndex: 2 }} />
      {label}
      {activated && <div className={styles.border} />}
    </div>
  )
}

Tab.propTypes = {
  style: PropTypes.object,
  className: PropTypes.string,
  label: PropTypes.string,
  activated: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
}

Tab.defaultProps = {
  style: {},
  className: '',
  label: '',
  activated: false,
}

export default onlyUpdateForPropTypes(Tab)
