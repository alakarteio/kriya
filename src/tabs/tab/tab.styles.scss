@import '~styles/theme';

.tab {
  align-items: center;
  cursor: pointer;
  display: flex;
  flex: 1;
  justify-content: center;
  padding: 1em;
  position: relative;

  &.activated {
    font-weight: bold;
  }
}

.border {
  background-color: $primary-color;
  bottom: 0;
  height: .5em;
  left: 0;
  position: absolute;
  width: 100%;
}
