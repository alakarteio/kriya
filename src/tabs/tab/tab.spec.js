/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import mockComponent from '__mocks__/component'
import Tab from './tab'
import { mapStateToProps, mapDispatchToProps } from './tab.container'

jest.mock('react-ink', () => mockComponent('ink'))

const snapshot = model => expect(model).toMatchSnapshot()
const snap = (props) => {
  const component = renderer.create(
    <Tab onClick={() => {}} {...props} />,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/tabs/tab', () => {
  describe('graphical (JSX)', () => {
    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should have a default behaviour', () => snap({}))
    it('should pass a label', () => snap({ label: 'a label' }))
    it('should activate the tab', () => snap({ activated: true }))
  })

  describe('container', () => {
    const state = {
      ui: {
        tabs: {
          activated: 'FIND',
          data: {
            CHOOSE: {
              code: 'CHOOSE',
              label: 'A choose label',
              an: 'other choose property',
            },
            FIND: {
              code: 'FIND',
              label: 'A find label',
              an: 'other find property',
            },
          },
        },
      },
    }

    it('should maps isActivated and tabs information', () => snapshot(mapStateToProps(state, { code: 'FIND' })))
    it('should map an isActivated to false', () => snapshot(mapStateToProps(state, { code: 'CHOOSE' })))
    it('should dispatch a TAB_CLICKED event', () => {
      const dispatch = jest.fn()
      const props = mapDispatchToProps(dispatch, { code: 'FIND' })

      props.onClick()

      expect(dispatch.mock.calls.length).toBe(1)
      expect(dispatch.mock.calls[0]).toEqual([{ type: 'TAB_CLICKED', payload: 'FIND' }])
    })
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
