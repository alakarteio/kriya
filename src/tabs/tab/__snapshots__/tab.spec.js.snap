// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/tabs/tab container should map an isActivated to false 1`] = `
Object {
  "activated": false,
  "an": "other choose property",
  "code": "CHOOSE",
  "label": "A choose label",
}
`;

exports[`common/tabs/tab container should maps isActivated and tabs information 1`] = `
Object {
  "activated": true,
  "an": "other find property",
  "code": "FIND",
  "label": "A find label",
}
`;

exports[`common/tabs/tab graphical (JSX) should activate the tab 1`] = `
<div
  className="tab activated"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="ink"
  >
    {"name":"ink","props":{"style":{"zIndex":2}}}
  </div>
  
  <div
    className="border"
  />
</div>
`;

exports[`common/tabs/tab graphical (JSX) should add custom className 1`] = `
<div
  className="tab custom"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="ink"
  >
    {"name":"ink","props":{"style":{"zIndex":2}}}
  </div>
  
</div>
`;

exports[`common/tabs/tab graphical (JSX) should add custom style 1`] = `
<div
  className="tab"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <div
    className="ink"
  >
    {"name":"ink","props":{"style":{"zIndex":2}}}
  </div>
  
</div>
`;

exports[`common/tabs/tab graphical (JSX) should have a default behaviour 1`] = `
<div
  className="tab"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="ink"
  >
    {"name":"ink","props":{"style":{"zIndex":2}}}
  </div>
  
</div>
`;

exports[`common/tabs/tab graphical (JSX) should pass a label 1`] = `
<div
  className="tab"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="ink"
  >
    {"name":"ink","props":{"style":{"zIndex":2}}}
  </div>
  a label
</div>
`;
